test/diExports.ts

```diff
@@ -1,13 +1,19 @@
+// import { TEST_BROWSER_DYNAMIC_APPLICATION_PROVIDERS } from '@angular/platform-browser-dynamic/testing';
+// import { TEST_BROWSER_DYNAMIC_PLATFORM_PROVIDERS } from '@angular/platform-browser-dynamic/testing';
 import { provide, Type }                              from '@angular/core';
 import { ComponentFixture, TestComponentBuilder }     from '@angular/compiler/testing';
-import { inject, async }                              from '@angular/core/testing';
+import { inject, async, setBaseTestProviders}                              from '@angular/core/testing';
 import { disableDeprecatedForms, provideForms, FormControl } from '@angular/forms';
 import { App, Config, Form, NavController, Platform } from 'ionic-angular';
 import { ConfigMock, NavMock }                        from './mocks';
 import { Utils }                                      from '../app/services/utils';
 export { TestUtils }                                  from './testUtils';
 
+// Your patch suggest I use setBaseTestProviders here?
```

Comment:
@lathonez  your patch suggests I use setBaseTestProviders here instead of app.spec.ts.
I get this error

Failed: addProviders can't be called after the injector has been already cr
eated for this test. This is most likely because you've already used the injector t
o inject a beforeEach or the current `it` function.                                
        Error: addProviders can't be called after the injector has been already cre
ated for this test. This is most likely because you've already used the injector to
 inject a beforeEach or the current `it` function.